packages/aws-cdk-lib/aws-ecs/lib/cluster.ts

```diff
@@ -1244,6 +1252,13 @@ export class AsgCapacityProvider extends Construct {
         throw new Error(`Invalid Capacity Provider Name: ${props.capacityProviderName}, If a name is specified, it cannot start with aws, ecs, or fargate.`);
       }
     }
+
+    if (props.instanceWarmupPeriod) {
```

Comment:
Just in case,It would be good to check whether it is Token.

```suggestion
    if (props.instanceWarmupPeriod && !Token.isUnresolved(props.instanceWarmupPeriod)) {
```